gensim/models/nmf.py

```diff
@@ -0,0 +1,643 @@
+        formatted : bool, optional
+            Whether the topic representations should be formatted as strings. If False, they are returned as
+            2 tuples of (word, probability).
+
+        Returns
+        -------
+        list of {str, tuple of (str, float)}
+            a list of topics, each represented either as a string (when `formatted` == True) or word-probability
+            pairs.
+
+        """
+        # TODO: maybe count sparsity in some other way
+        sparsity = self._W.getnnz(axis=0)
+
+        if num_topics < 0 or num_topics >= self.num_topics:
+            num_topics = self.num_topics
+            chosen_topics = range(num_topics)
+        else:
+            num_topics = min(num_topics, self.num_topics)
+
+            sorted_topics = list(matutils.argsort(sparsity))
+            chosen_topics = (
+                sorted_topics[: num_topics // 2] + sorted_topics[-num_topics // 2:]
+            )
+
+        shown = []
+
+        topics = self.get_topics()
+
+        for i in chosen_topics:
+            topic = topics[i]
+            bestn = matutils.argsort(topic, num_words, reverse=True).ravel()
+            topic = [(self.id2word[id], topic[id]) for id in bestn]
+            if formatted:
+                topic = " + ".join(['%.3f*"%s"' % (v, k) for k, v in topic])
+
+            shown.append((i, topic))
+            if log:
+                logger.info("topic #%i (%.3f): %s", i, sparsity[i], topic)
+
+        return shown
+
+    def show_topic(self, topicid, topn=10):
+        """Get the representation for a single topic. Words here are the actual strings, in constrast to
+        :meth:`~gensim.models.nmf.Nmf.get_topic_terms` that represents words by their vocabulary ID.
+
+        Parameters
+        ----------
+        topicid : int
+            The ID of the topic to be returned
+        topn : int, optional
+            Number of the most significant words that are associated with the topic.
+
+        Returns
+        -------
+        list of (str, float)
+            Word - probability pairs for the most relevant words generated by the topic.
+
+        """
+        return [
+            (self.id2word[id], value)
+            for id, value in self.get_topic_terms(topicid, topn)
+        ]
+
+    def get_topic_terms(self, topicid, topn=10):
+        """Get the representation for a single topic. Words the integer IDs, in constrast to
+        :meth:`~gensim.models.nmf.Nmf.show_topic` that represents words by the actual strings.
+
+        Parameters
+        ----------
+        topicid : int
+            The ID of the topic to be returned
+        topn : int, optional
+            Number of the most significant words that are associated with the topic.
+
+        Returns
+        -------
+        list of (int, float)
+            Word ID - probability pairs for the most relevant words generated by the topic.
+
+        """
+        topic = self.get_topics()[topicid]
```

Comment:
you can extract + normalize single row/col here, instead of converting the full internal matrix for single column